src/components/Board.js

```diff
@@ -4,19 +4,25 @@ import Square from './Square';
 import PropTypes from 'prop-types';
 
 
-const generateSquareComponents = (squares, onClickCallback) => {
+const generateSquareComponents = (squares) => {
   // Complete this for Wave 1
   // squares is a 2D Array, but 
   //  you need to return a 1D array
   //  of square components
-
+  const gridBoard = []; //grid needs to 3 by 3, set up a perimeter
+  for (let row = 0; row < 3; row += 1) {
+    for (let column = 0; column < 3; column += 1) {
+      gridBoard.push(squares[row][column]); //push ~= append
+    }
+  }
+  return gridBoard;
```

Comment:
Good use of a nested for loop to iterate through the 2d array.

Reply:
Currently, this for loop makes a 1D copy of the `squares` prop. Instead, you could use this for loop to create a `Squares` component for each item in the `squares` array.


 ```suggestion
  const gridBoard = []; //grid needs to 3 by 3, set up a perimeter
  for (let row = 0; row < 3; row += 1) {
    for (let column = 0; column < 3; column += 1) {
      gridBoard.push(<Square key={square.id} id={square.id} value={square.value} onClickCallback={onClickCallback} />); //push ~= append
    }
  }
  return gridBoard;
```